### PR TITLE
[release/9.0] Update Azure.Identity version

### PR DESCRIPTION
**Pull request by AndriySvyryd**

### CODE REVIEW

**Review by roji — APPROVED** [review]

**Review by SamMonoRT — APPROVED** [review]

File: eng/Versions.props [hunk 1]
```diff
@@ -39,7 +39,7 @@
     <!-- NB: This version affects Visual Studio compatibility. See https://learn.microsoft.com/visualstudio/extensibility/roslyn-version-support -->
     <MicrosoftCodeAnalysisVersion>4.8.0</MicrosoftCodeAnalysisVersion>
     <MicrosoftCodeAnalysisTestingVersion>1.1.2-beta1.24121.1</MicrosoftCodeAnalysisTestingVersion>
-    <AzureIdentityVersion>1.11.3</AzureIdentityVersion>
+    <AzureIdentityVersion>1.12.0</AzureIdentityVersion>
```
[the remark this answers is not in the document]
Comment (AndriySvyryd) [reply eng/Versions.props:42]: At least as often as new CVEs are found.
Most repos handle non-dotnet dependency updates manually.

[the remark this answers is not in the document]
Comment (AndriySvyryd) [reply eng/Versions.props:42]: Dependabot still doesn't work that well for arcade-style repos https://github.com/dependabot/dependabot-core/issues/8490

[the remark this answers is not in the document]
Comment (ErikEJ) [reply eng/Versions.props:42]: Does this apply to the Cosmos or SQL Server provider?

[the remark this answers is not in the document]
Comment (AndriySvyryd) [reply eng/Versions.props:42]: And this is only for tests

**Pull request merged**